File: Bit Manipulation/flipBitToWin.cpp
#include <stdio.h>
#include <iostream>

using namespace std;

unsigned long flipBitToWin(unsigned long a){
  int count=0;
  int maxLength=0;
  int count2=0;
  bool secondtry=false;
  while(a){
    if(a&1) count2++;
    else{
      if(a&2){
        count=count2;
        count2=1;
      }else{
        if(count+count2>maxLength) maxLength=count+count2;
        count=0;
        count2=0;
      }
    }
    if(count+count2>maxLength) maxLength=count+count2;
    a>>=1;
  }
  return maxLength;
}

int main(){
  unsigned long a=14815;
  printf("%lu\n",flipBitToWin(a));
}

File: Bit Manipulation/flipBitToWin.java
public class flipBitToWin{

  static long flipBitToWin(long a){
    int prev=0;
    long maxLength=0;
    int curr=0;
    while(a!=0){
      if((a&1)==1) curr++;
      else{
        if((a&2)==2){
          prev=curr+1;
          curr=0;
        }else{
          if(prev+curr>maxLength) maxLength=prev+curr;
          prev=0;
          curr=0;
        }
      }
      if(prev+curr>maxLength) maxLength=prev+curr;
      a>>=1;
    }
    return maxLength;
  }

  public static void main(String[] args){
    long a=14815;
    System.out.println(flipBitToWin(a));
  }

}
